test/app/exercises_test.rb

```diff
@@ -9,11 +9,27 @@ def app
     ExercismWeb::App
   end
 
-  def test_excercises_by_key
-    alice = User.create(username: 'alice', github_id: 1)
-    exercise = UserExercise.create(user: alice, language: 'go', slug: 'one')
-    submission = Submission.create(user: alice, language: 'go', slug: 'bob', user_exercise: exercise)
+  attr_reader :alice, :exercise, :submission
+
+  def setup
+    super
+    @alice = User.create(username: 'alice', github_id: 1)
+    @exercise = UserExercise.create(user: alice, language: 'go', slug: 'one')
+    @submission = Submission.create(user: alice, language: 'go',
+                                    slug: 'bob', user_exercise: exercise)
+  end
+
+  def test_exercises_next_with_session_values
+    get "/exercises/next", {}, login(alice)
+    assert_equal 302, last_response.status
```

Comment:
This doesn't have the session values for inbox. We should probably check that it goes to the correct page based on alice's default.